pkg/asset/ignition/bootstrap/content/report.go

```diff
@@ -0,0 +1,58 @@
+package content
+
+const (
+	// ReportSystemdContents is a service that reports the bootstrap progress
+	// via a Kubernetes Event.
+	ReportSystemdContents = `
+[Unit]
+Description=Report the completion of the cluster bootstrap process
+# Workaround for https://github.com/systemd/systemd/issues/1312
+Wants=bootkube.service tectonic.service
+After=bootkube.service tectonic.service
+
+[Service]
+# Workaround for https://github.com/systemd/systemd/issues/1312 and https://github.com/opencontainers/runc/pull/1807
+ExecStartPre=/usr/bin/test -f /opt/tectonic/.bootkube.done
+ExecStartPre=/usr/bin/test -f /opt/tectonic/.tectonic.done
+ExecStart=/opt/tectonic/report-progress.sh /opt/tectonic/auth/kubeconfig bootstrap-complete "cluster bootstrapping has completed"
+
+Restart=on-failure
+RestartSec=5s
+
+[Install]
+WantedBy=multi-user.target
+`
+
+	// ReportShFileContents is a script for reporting the bootstrap progress.
+	ReportShFileContents = `#!/usr/bin/env bash
+set -e
+
+KUBECONFIG="${1}"
+NAME="${2}"
+MESSAGE="${3}"
+
+echo "Reporting install progress..."
+
+manifest=$(mktemp)
+trap "rm ${manifest}" EXIT
+
+cat <<EOF > "${manifest}"
+apiVersion: v1
+kind: Event
+metadata:
+  name: "${NAME}"
+  namespace: kube-system
+involvedObject:
+  namespace: kube-system
+message: "${MESSAGE}"
+firstTimestamp: $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+lastTimestamp: $(date -u +'%Y-%m-%dT%H:%M:%SZ')
```

Comment:
Call `date` once before this here-document:

```sh
DATE="$(date -u +'%Y-%m-%dT%H:%M:%SZ')
```

and then use that date for both of these?

```
firstTimestamp: "${DATE}"
lastTimestamp: "${DATE}"
```

Saving a second `date` call is probably not going to be a noticeable improvement, but by using the same value in both places folks won't have to worry about whether the small difference between the successive calls is meaningful.